### PR TITLE
[RFR] Convert filter components to useStyles

### PR DESCRIPTION
**Pull request by m4theushw**
Related to #3453 
- [x] ra-ui-materialui/src/list/Filter.js
- [X] ra-ui-materialui/src/list/FilterButton.js
- [X] ra-ui-materialui/src/list/FilterButtonMenuItem.js
- [x] ra-ui-materialui/src/list/FilterForm.js

### CODE REVIEW

File: packages/ra-ui-materialui/src/list/FilterButton.js [hunk 1]
```diff
@@ -1,122 +1,103 @@
...
-        this.handleClickButton = this.handleClickButton.bind(this);
-        this.handleRequestClose = this.handleRequestClose.bind(this);
-        this.handleShow = this.handleShow.bind(this);
-    }
+const FilterButton = ({
+    showFilter,
+    filters,
+    displayedFilters,
+    filterValues,
+    classes = {},
```
Comment (djhi) [inline packages/ra-ui-materialui/src/list/FilterButton.js]: rename it `classesOverride` like this:
`classes: classesOverride`

File: packages/ra-ui-materialui/src/list/FilterButton.js [hunk 2]
```diff
@@ -1,122 +1,103 @@
...
+    className,
+    resource,
+    ...rest
+}) => {
+    const [open, setOpen] = useState(false);
+    const [anchorEl, setAncholEl] = useState();
+    const styles = useStyles();
```
Comment (djhi) [inline packages/ra-ui-materialui/src/list/FilterButton.js]:
```suggestion
    const styles = useStyles({ classes: classesOverride });
```
Allows users to override classes

File: packages/ra-ui-materialui/src/list/FilterButton.js [hunk 3]
```diff
@@ -1,122 +1,103 @@
-import React, { Component } from 'react';
+import React, { Component, useState, useCallback, useMemo } from 'react';
 import PropTypes from 'prop-types';
 import Menu from '@material-ui/core/Menu';
-import { withStyles, createStyles } from '@material-ui/core/styles';
+import { withStyles, makeStyles } from '@material-ui/core/styles';
```
Comment (djhi) [inline packages/ra-ui-materialui/src/list/FilterButton.js]: withStyles is not needed anymore

File: packages/ra-ui-materialui/src/list/FilterButton.js [hunk 4]
```diff
@@ -1,122 +1,103 @@
...
 import ContentFilter from '@material-ui/icons/FilterList';
 import classnames from 'classnames';
 import compose from 'recompose/compose';
```
Comment (djhi) [inline packages/ra-ui-materialui/src/list/FilterButton.js]: not needed anymore

File: packages/ra-ui-materialui/src/list/FilterForm.js [hunk 5]
```diff
@@ -69,53 +69,64 @@ const sanitizeRestProps = ({
...
+            filters.filter(
+                filterElement =>
+                    filterElement.props.alwaysOn ||
+                    displayedFilters[filterElement.props.source] ||
+                    typeof lodashGet(
+                        initialValues,
+                        filterElement.props.source
+                    ) !== 'undefined'
+            ),
+        [displayedFilters, filters, initialValues]
```
Comment (fzaninotto) [inline packages/ra-ui-materialui/src/list/FilterForm.js]: `filters` changes on every render, so the `useMemo` here is useless.

Comment (m4theushw) [reply packages/ra-ui-materialui/src/list/FilterForm.js]: The `useEffect` [here](https://github.com/marmelab/react-admin/blob/0282a34e3869838d0aebf9a40ca899bc41d991bf/packages/ra-ui-materialui/src/list/FilterForm.js#L82) would be useless too?

Comment (fzaninotto) [reply packages/ra-ui-materialui/src/list/FilterForm.js]: yes, that `useEffect` should use an empty array as second argument

**Comment by fzaninotto** [comment]
This PR now needs rebase

**Comment by m4theushw** [comment]
Changes rebased!

**Comment by fzaninotto** [comment]
I've fonverted 3 out of 4 of these components in another PR. Can you rebase so that your PR only converts FilterForm?

**Comment by m4theushw** [comment]
Yes, PR updated.

**Comment by fzaninotto** [comment]
Superseded by #3732

**Pull request closed without merging**